src/jit/codegenxarch.cpp

```diff
@@ -5279,86 +5325,161 @@ void CodeGen::genConsumePutStructArgStk(
 }
 #endif // FEATURE_UNIX_AMD64_STRUCT_PASSING
 
-void CodeGen::genConsumeBlockOp(GenTreeBlkOp* blkNode, regNumber dstReg, regNumber srcReg, regNumber sizeReg)
+//------------------------------------------------------------------------
+// genConsumeBlockSize: Ensure that the block size is in the given register
+//
+// Arguments:
+//    blkNode - The block node
+//    sizeReg - The register into which the block's size should go
+//
+
+void CodeGen::genConsumeBlockSize(GenTreeBlk* blkNode, regNumber sizeReg)
+{
+    unsigned blockSize = blkNode->Size();
+    if (sizeReg != REG_NA)
+    {
+        if (blockSize != 0)
+        {
+            assert(blkNode->gtRsvdRegs == genRegMask(sizeReg));
+            genSetRegToIcon(sizeReg, blockSize);
+        }
+        else
+        {
+            noway_assert(blkNode->gtOper == GT_STORE_DYN_BLK);
+            genConsumeRegAndCopy(blkNode->AsDynBlk()->gtDynamicSize, sizeReg);
```

Comment:
Give the massive amount of changes, I cannot claim to have reviewed each line of code.  Based on what I have reviewed looks good to me.